tutorials/dataframe/df027_SQliteDependencyOverVersion.C

```diff
@@ -13,47 +13,50 @@
 /// \macro_image
 ///
 /// \date August 2018
-/// \author Alexandra-Maria Dobrescu
+/// \authors Alexandra-Maria Dobrescu, Sergey Linev
 
-void df027_SQliteDependencyOverVersion () {
 
+void df027_SQliteDependencyOverVersion ()
+{
    auto rdfb = ROOT::RDF::FromSqlite("http://root.cern/files/root_download_stats.sqlite", "SELECT * FROM accesslog;");
 
    auto minTimeStr = *rdfb.Reduce([](std::string a, std::string b) {return std::min(a, b);}, "Time", std::string("Z"));
 
    std::cout << "Minimum time is '" << minTimeStr << "'" << std::endl;
 
    double minTime = TDatime(minTimeStr.c_str()).Convert();
-   double now = TDatime().Convert();
+   double maxTime = minTime + 3600.*24*365.25*4; // cover approx 4 years from minimal time
 
    auto rdf = rdfb.Define("datime", [](const std::string &time){return TDatime(time.c_str()).Convert();}, {"Time"});
 
    auto h614 = rdf.Filter([](const std::string &v){ return 0 == v.find("6.14");}, {"Version"})
-                  .Histo1D({"h614", "Download time for version 6.14", 16, minTime, now}, {"datime"});
+                  .Histo1D({"h614", "Download time for version 6.14", 64, minTime, maxTime}, {"datime"});
 
    auto h616 = rdf.Filter([](const std::string &v){ return 0 == v.find("6.16");}, {"Version"})
-                  .Histo1D({"h616", "Download time for version 6.16", 16, minTime, now}, {"datime"});
+                  .Histo1D({"h616", "Download time for version 6.16", 64, minTime, maxTime}, {"datime"});
 
    auto h618 = rdf.Filter([](const std::string &v){ return 0 == v.find("6.18");}, {"Version"})
-                  .Histo1D({"h618", "Download time for version 6.18", 16, minTime, now}, {"datime"});
-
-   // Add here a newer version!
+                  .Histo1D({"h618", "Download time for version 6.18", 64, minTime, maxTime}, {"datime"});
 
    auto histoList = {h614, h616, h618};
-   auto canvases = new std::vector<TCanvas*>(histoList.size());
 
-   gStyle->SetTimeOffset(0);
-   gStyle->SetOptStat(0);
-   auto histoIdx = 0U;
+   std::cout << "Be aware that rdf.Filter() execution takes ~30s" << std::endl;
+
+   std::vector<TObject*> drawHistos;
+
    for (auto histo : histoList) {
-      canvases->at(histoIdx) = new TCanvas();
-      histo->GetXaxis()->LabelsOption("v");
       histo->GetXaxis()->SetTimeDisplay(1);
       histo->GetXaxis()->SetLabelSize(0.02);
       histo->GetXaxis()->SetNdivisions(512, kFALSE);
-      histo->GetXaxis()->SetTimeFormat("%Y-%m-%d");
-      histo->DrawClone();
-      histoIdx++;
+      histo->GetXaxis()->SetTimeFormat("%Y-%m-%d%F1970-00-00 00:00:00");
+      histo->SetStats(kFALSE);
+
+      drawHistos.emplace_back(histo->Clone());
    }
 
+   auto c1 = new TCanvas("c1","Download time", 800, 1500);
+   c1->Divide(1, drawHistos.size());
+   for (unsigned n = 0; n < drawHistos.size(); ++n)
+      c1->GetPad(n+1)->Add(drawHistos[n]);
+
```

Comment:
Just for me to understand, where is the `Draw` part here?

Reply:
> where is the Draw part here

`c1->GetPad(n+1)->Add(drawHistos[n]);`

Reply:
I don't understand then. The `TPad::Add` method does two things? Both adds a new drawable object to the list of drawables that the canvas is responsible for **and** draws the canvas at the same time? Is the canvas redrawn (updated) for every call to `TPad::Add`?

Reply:
>  Is the canvas redrawn (updated) for every call to TPad::Add?

No, canvas marked as modified each time new object add. 
At the end of macro execution ROOT call `c->Update()` for all modified canvases.